python/core/layertree/qgslayertree.sip

```diff
@@ -1,61 +1,193 @@
+/************************************************************************
+ * This file has been generated automatically from                      *
+ *                                                                      *
+ * src/core/layertree/qgslayertree.h                                    *
+ *                                                                      *
+ * Do not edit manually ! Edit header and run scripts/sipify.pl again   *
+ ************************************************************************/
+
+
+
 class QgsLayerTree : QgsLayerTreeGroup
 {
-%TypeHeaderCode
-#include <qgslayertree.h>
+%Docstring
+ Namespace with helper functions for layer tree operations.
+
+ Only generally useful routines should be here. Miscellaneous utility functions for work
+ with the layer tree are in QgsLayerTreeUtils class.
+
+.. versionadded:: 3.0
 %End
 
+%TypeHeaderCode
+#include "qgslayertree.h"
+%End
   public:
+
     static bool isGroup( QgsLayerTreeNode *node );
+%Docstring
+ Check whether the node is a valid group node
+
+.. versionadded:: 2.4
+ :rtype: bool
+%End
+
     static bool isLayer( const QgsLayerTreeNode *node );
+%Docstring
+ Check whether the node is a valid layer node
+
+.. versionadded:: 2.4
+ :rtype: bool
+%End
 
-    //! Cast node to a group. No type checking is done - use isGroup() to find out whether this operation is legal.
-    // PYTHON: automatic cast
-    //inline QgsLayerTreeGroup* toGroup( QgsLayerTreeNode* node );
 
-    //! Cast node to a layer. No type checking is done - use isLayer() to find out whether this operation is legal.
-    // PYTHON: automatic cast
-    //inline QgsLayerTreeLayer* toLayer( QgsLayerTreeNode* node );
+
+
+    QgsLayerTree();
```

Comment:
The default constructor should be in SIP as well - I think SIP creates a default constructor if no other constructor is specified.

The copy constructor actually should be probably protected or private in the header - we have clone() method instead...